File: examples/filter-by-state.py
#!/usr/bin/env python3
# Filter fireworks events by state
# Usage: python examples/filter-by-state.py TX
import json
import sys
from pathlib import Path
from datetime import datetime, timezone

data_path = Path(__file__).parent.parent / 'data' / 'events' / '2026-us.json'
events = json.loads(data_path.read_text())

if len(sys.argv) < 2:
    print('Usage: python filter-by-state.py <STATE>', file=sys.stderr)
    print('Example: python filter-by-state.py TX', file=sys.stderr)
    sys.exit(1)

state = sys.argv[1].upper()
results = [e for e in events if e.get('state') == state]
print(f'{len(results)} events in {state}:\n')
for e in results:
    dt = datetime.fromisoformat(e['startDate']).strftime('%b %-d')
    print(f"  {dt} — {e['title']} ({e['city']}) [{e['status']}]")

File: examples/filter-by-state.js
// Filter fireworks events by state (Node.js ESM)
// Usage: node examples/filter-by-state.js TX
import { readFileSync } from 'node:fs'
import { resolve, dirname } from 'node:path'
import { fileURLToPath } from 'node:url'

const __dirname = dirname(fileURLToPath(import.meta.url))
const events = JSON.parse(
  readFileSync(resolve(__dirname, '../data/events/2026-us.json'), 'utf8')
)

const state = process.argv[2]?.toUpperCase()
if (!state) {
  console.error('Usage: node filter-by-state.js <STATE>')
  console.error('Example: node filter-by-state.js TX')
  process.exit(1)
}

const results = events.filter(e => e.state === state)
console.log(`${results.length} events in ${state}:\n`)
for (const e of results) {
  const date = new Date(e.startDate).toLocaleDateString('en-US', { month: 'short', day: 'numeric' })
  console.log(`  ${date} — ${e.title} (${e.city}) [${e.status}]`)
}
